src/main/kotlin/no/nav/familie/ef/sak/beregning/skolepenger/BeregningSkolepengerService.kt

```diff
@@ -1,33 +1,97 @@
+    fun beregnYtelse(utgiftsperioder: List<UtgiftsperiodeSkolepengerDto>): BeregningSkolepengerResponse {
         validerGyldigePerioder(utgiftsperioder)
         validerFornuftigeBeløp(utgiftsperioder)
 
-        return utgiftsperioder.map {
-            BeløpsperiodeSkolepengerDto(
-                it.tilPeriode(),
-                beløp = 100, // TODO håndtere senere
-                BeregningsgrunnlagSkolepengerDto(it.studietype, it.utgifter, it.studiebelastning)
-            )
+        // fra forrige vedtak
+        val tidligereForbruktePerioder = mutableMapOf<Year, Int>()
+
+        val perioder = utgiftsperioder.groupBy {
+            Year.of(if (it.årMånedFra.monthValue > 6) it.årMånedFra.year else it.årMånedFra.year.minus(1))
+        }.map {
+            val previous = tidligereForbruktePerioder.getOrDefault(it.key, 0)
+            val periode = it.value.singleOrNull()
+            brukerfeilHvis(periode == null) {
+                "Antall perioder for skoleår=${it.key} er fler enn 1"
+            }
+            beregn(it.key, previous, periode)
         }
+        return BeregningSkolepengerResponse(perioder)
+    }
+
+    private fun beregn(
+        skoleår: Year,
+        tidligereForbrukt: Int,
+        periode: UtgiftsperiodeSkolepengerDto
+    ): BeløpsperiodeSkolepenger {
+        val maksbeløp = maksbeløpPerSkoleår
+        var nyForbrukt = tidligereForbrukt
+        val studiebelastning = periode.studiebelastning.toBigDecimal()
+        val antallMåneder = minOf(MONTHS.between(periode.årMånedFra, periode.årMånedTil) + 1, 10)
+        feilHvis(antallMåneder < 1 || antallMåneder > 10) {
+            "Antall måneder er $antallMåneder men må være 1-10"
+        }
+        val månedsdel = antallMåneder.toBigDecimal().divide(BigDecimal.TEN)
+        val maksbeløpFordeltAntallMåneder = maksbeløp.toBigDecimal().multiply(månedsdel).roundUp()
+        val maksbeløpEtterStudieredusering =
+            maksbeløpFordeltAntallMåneder.multiply(studiebelastning.divide(ONE_HUNDRED)).roundUp().toInt()
+        val tilgjengeligFraTidligere = maxOf(maksbeløp - nyForbrukt, 0)
+        var tilgjengelig = minOf(tilgjengeligFraTidligere, maksbeløpEtterStudieredusering)
+        val nyeUtbetalinger = periode.utgifter.map {
+            val nyTilgjengelig = maxOf(tilgjengelig - it.utgifter, 0)
+            val utbetales = tilgjengelig - nyTilgjengelig
+            tilgjengelig = nyTilgjengelig
+            nyForbrukt += utbetales
+            BeregnetUtbetalingSkolepenger(utbetales, it)
+        }
+        return BeløpsperiodeSkolepenger(
+            skoleår = skoleår,
+            maksbeløp = maksbeløp,
+            maksbeløpFordeltAntallMåneder = maksbeløpEtterStudieredusering,
+            alleredeUtbetalt = tidligereForbrukt,
+            nyForbrukt = nyForbrukt, // Burde denna være en løpende nyForbrukt fra forrige periode?
+            utbetalinger = nyeUtbetalinger,
+            grunnlag = BeregningsgrunnlagSkolepengerDto(
```

Comment:
Istedenfor å "regne ut" hva som skal utbetales så tror jeg vi bør bruke `stønad` direkte og heller gjøre en validering på at `stønad` summert innenfor en periode ikke overstiger maksbeløp for skoleåret - og kaste feil hvis det oppstår. Usikker på om vi skal prøve oss på å beregne MED prosentsats og antall måneder - og heller la saksbehandler styre den jobben?